src/SFA.DAS.AssessorService.Data/IlrRepository.cs

```diff
@@ -52,5 +55,52 @@ public async Task<IEnumerable<Ilr>> Search(string searchQuery)
             return await _context.Ilrs.Where(r => r.FamilyName == searchQuery || r.GivenNames == searchQuery || r.Uln == long.Parse(searchQuery))
                 .ToListAsync();
         }
+
+        public async Task RefreshFromSubmissionEventData(Guid id, SubmissionEvent subEvent)
+        {
+            var learnerRecord = await _context.Ilrs.FirstAsync(i => i.Id == id);
+            if(learnerRecord.EventId != subEvent.Id) learnerRecord.EventId = subEvent.Id;
+            if(learnerRecord.UkPrn != (int)subEvent.Ukprn) learnerRecord.UkPrn = (int)subEvent.Ukprn;
+            if(learnerRecord.FamilyName != subEvent.FamilyName) learnerRecord.FamilyName = subEvent.FamilyName;
+            if(learnerRecord.ApprenticeshipId != subEvent.ApprenticeshipId) learnerRecord.ApprenticeshipId = subEvent.ApprenticeshipId;
+            if(learnerRecord.GivenNames != subEvent.GivenNames) learnerRecord.GivenNames = subEvent.GivenNames;
+            if(learnerRecord.EpaOrgId != subEvent.EPAOrgId) learnerRecord.EpaOrgId = subEvent.EPAOrgId;
+            if (learnerRecord.CompletionStatus != subEvent.CompStatus)
+                learnerRecord.CompletionStatus = subEvent.CompStatus;
+            if (learnerRecord.PlannedEndDate != subEvent.PlannedEndDate)
+            {
+                if (subEvent.PlannedEndDate.HasValue)
+                    learnerRecord.PlannedEndDate = subEvent.PlannedEndDate.Value;
+            }
+            learnerRecord.Source = ContactStatus.Live;
+            learnerRecord.UpdatedAt = DateTime.Now;
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task MarkAllUpToDate(long uln)
+        {
+            var learnerRecords = await _context.Ilrs.Where(i => i.Uln == uln).ToListAsync();
+            foreach (var learnerRecord in learnerRecords)
+            {
+                learnerRecord.UpdatedAt = DateTime.Now;
+            }
+            await _context.SaveChangesAsync();
+        }
+        private string createILRQuery(string filter)
+        {
+            return @"SELECT ilr.*
+                 FROM
+	                (
+		                SELECT *,
+		                ROW_NUMBER() OVER (PARTITION BY [Uln], [StdCode] ORDER BY [EventId] DESC, [Source] DESC, [CreatedAt] DESC, [LearnStartDate] DESC ) AS rownumber
```

Comment:
Am I right in thinking we no longer need DENSE_RANK() ?

Reply:
Yes we don't need DENSE_RANK anymore.